homeassistant/components/faadelays/binary_sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Platform for FAA Delays sensor component."""
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import ATTR_ICON, ATTR_NAME
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, FAA_BINARY_SENSORS
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up a FAA sensor based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors = []
+    for kind, attrs in FAA_BINARY_SENSORS.items():
+        name = attrs[ATTR_NAME]
+        icon = attrs[ATTR_ICON]
+
+        binary_sensors.append(
+            FAABinarySensor(coordinator, kind, name, icon, entry.entry_id)
+        )
+
+    async_add_entities(binary_sensors, True)
+
+
+class FAABinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """Define a binary sensor for FAA Delays."""
+
+    def __init__(self, coordinator, sensor_type, name, icon, entry_id):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        self.coordinator = coordinator
+        self._entry_id = entry_id
+        self._icon = icon
+        self._name = name
+        self._sensor_type = sensor_type
+        self._state = None
+        self._id = self.coordinator.data.iata
+        self._attrs = {}
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._id} {self._name}"
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def state(self):
```

Comment:
Binary sensors should not implement the `state` property. Implement `is_on`.

https://developers.home-assistant.io/docs/core/entity/binary-sensor#properties